presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -538,6 +539,22 @@ public void testIsNotNullWithNestedData()
         assertUpdate(admin, "DROP TABLE nest_test");
     }
 
+    @Test
+    public void testNestedDataExceedsMaxColumnsOutput()
```

Comment:
since this is an engine feature, would it be better if we test it in presto-main somehow?

Reply:
I added a test in presto-main as well.